sympy/functions/elementary/hyperbolic.py

```diff
@@ -336,7 +336,7 @@ def _eval_as_leading_term(self, x):
             return self.func(arg)
 
     def _eval_is_real(self):
-        return self.args[0].is_real
+        return self.args[0].is_real or self.args[0].is_imaginary
```

Comment:
Assuming ``x`` and ``y`` real,
`` cosh(x + iy) = cosh(x)cos(y) + i sinh(x)sin(y)`` is real
when ``sinh(x) == 0`` or ``sin(y) == 0``
or equivalently, ``x == 0`` or ``y`` is a multiple of ``pi``.